virocon/distributions.py

```diff
@@ -1494,3 +1495,143 @@ def _fit_mle(self, sample):
 
     def _fit_lsq(self, data, weights):
         raise NotImplementedError()
+
+
+class ScipyDistribution(Distribution):
+    """
+    A base class used for virocon distributions derived from a scipy distribution.
+
+    To create a new virocon distribution inherit from ScipyDistribution
+    and overwrite either `scipy_dist` or `scipy_dist_name`.
+
+    Attributes
+    ----------
+    scipy_dist : sts.rv_continuous
+        A continuous distribution defined in scipy.
+        (Or acting like a distribution defined there.)
+    scipy_dist_name : str
+        The name of a continuous distribution defined in scipy.
+    """
+
+    scipy_dist: sts.rv_continuous
+    scipy_dist_name: str
+
+    def __init__(self, *args, **kwargs):
+        try:
+            self.scipy_dist = getattr(sts, self.scipy_dist_name)
+        except AttributeError:
+            pass
+        try:
+            self.scipy_dist
+        except AttributeError:
+            raise TypeError(
+                f"Can't instantiate abstract class {self.__class__.__name__}. "
+                "Overwrite either scipy_dist or scipy_dist_name in subclass."
+            )
+        self.scipy_dist_name = self.scipy_dist.name
+        self._param_names = self._list_scipy_parameters(self.scipy_dist)
+        self._set_default_parameter_values()
+
+        # read parameter from args: (shape(s), loc, scale)
+        assert len(args) <= len(self._param_names)
+        for arg, par_name in zip(args, self._param_names):
+            setattr(self, par_name, arg)
+
+        assert len(kwargs) <= 2 * len(self._param_names)
+        for key, arg in kwargs.items():
+            if key in self._param_names:
+                setattr(self, key, arg)
+            elif key.startswith("f_") and key[2:] in self._param_names:
+                setattr(self, key, arg)
+                setattr(self, key[2:], arg)
+            else:
+                raise TypeError(
+                    f"{self.__class__.__name__} got an unexpected keyword argument '{key}'"
+                )
+
+    #  https://stackoverflow.com/a/53640468
+    @staticmethod
+    def _list_scipy_parameters(distribution):
+        """List parameters for scipy.stats.distribution.
+        # Arguments
+            distribution: a string or scipy.stats distribution object.
+        # Returns
+            A list of distribution parameter strings.
+        """
+        if isinstance(distribution, str):
+            distribution = getattr(sts, distribution)
+        if distribution.shapes:
+            parameters = [name.strip() for name in distribution.shapes.split(",")]
+        else:
+            parameters = []
+        # assert distribution is continous
```

Comment:
How does the next line ( parameters += ["loc", "scale"] ) assert that the distribution is continuous? Or is the comment wrong?

Reply:
Removed